db/rdbms/src/main/java/io/camunda/db/rdbms/read/service/VariableReader.java

```diff
@@ -37,10 +40,14 @@ public VariableEntity findOne(final Long key) {
         .getFirst();
   }
 
-  public SearchResult search(final VariableQuery filter) {
-    LOG.trace("[RDBMS DB] Search for variables with filter {}", filter);
-    final var totalHits = variableMapper.count(filter);
-    final var hits = variableMapper.search(filter);
+  public SearchResult search(final VariableQuery query) {
+    final var dbSort = convertSort(query.sort(), VariableSearchColumn.VAR_KEY);
+    final var dbQuery =
+        VariableDbQuery.of(
+            b -> b.filter(query.filter()).sort(dbSort).page(convertPaging(dbSort, query.page())));
+    LOG.trace("[RDBMS DB] Search for variables with filter {}", query);
+    final var totalHits = variableMapper.count(dbQuery);
+    final var hits = variableMapper.search(dbQuery);
     return new SearchResult(hits, totalHits.intValue());
```

Comment:
❌   In other refactorings for the keyset pagination we also got rid of the local SearchResult class and used the SearchQueryResult directly (see ProcessInstanceReader). That one then also includes the sortValues.